components/SingleClassView/Attendance/AttendanceBox.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useContext, useState , useEffect } from "react";
+import styles from "./attendance.module.css";
+import { Attendance } from "../../../models/Attendance";
+import { CustomError } from "../../util/CustomError";
+import { CustomLoader } from "../../util/CustomLoader";
+import { AttendanceTypes, CreateAttendance } from "../,,/../../../models";
+import { AttendanceRow } from "./AttendanceRow";
+import { APIContext } from "../../../context/APIContext";
+
+
+type AttendanceBoxProps = {
+    attendances: Attendance[] | undefined,
+    sessionId: string,
+    classId: string,
+};
+
+const AttendanceBox: React.FC<AttendanceBoxProps> =  ({
+    attendances,
+    sessionId,
+    classId,
+}) => {
+    if (!attendances){
+        return <p>This date has no sessions</p>
+    }
+    const api = useContext(APIContext);
```

Comment:
this is kinda confusing, so lets clean up the variable names and add some comments as to what this section of code is doing